lib/linked_list.rb

```diff
@@ -12,69 +12,154 @@ def initialize(value, next_node = nil)
 
 # Defines the singly linked list
 class LinkedList
-    def initialize
+    def initialize 
       @head = nil # keep the head private. Not accessible outside this class
     end
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1) bc it only adds one thing regardless of how big the list is
+    # Space Complexity: O(1)
     def add_first(value)
-      raise NotImplementedError
+      if @head.nil? 
+        @head = Node.new(value)
+      else 
+        @head = Node.new(value, @head) 
+      end

```

Comment:
Could be simplified as:

```suggestion
      @head = Node.new(value, @head)
```